spirv_glsl.cpp

```diff
@@ -3221,6 +3239,199 @@ string CompilerGLSL::access_chain(uint32_t base, const uint32_t *indices, uint32
 	return expr;
 }
 
+string CompilerGLSL::access_chain(uint32_t base, const uint32_t *indices, uint32_t count, const SPIRType &target_type)
+{
+	if (flattened_buffer_blocks.count(base))
+	{
+		std::string expr;
+		flattened_access_chain(expr, base, indices, count, target_type, 0);
+
+		return expr;
+	}
+	else
+	{
+		return access_chain(base, indices, count, false);
+	}
+}
+
+void CompilerGLSL::flattened_access_chain(std::string &expr, uint32_t base, const uint32_t *indices, uint32_t count, const SPIRType &target_type, uint32_t offset)
+{
+	if (!target_type.array.empty())
+	{
+		SPIRV_CROSS_THROW("Access chains that result in an array can not be flattened");
```

Comment:
The major limitation here actually is just that legacy GLSL (at least on GLES I think?) does not support array literals - the natural translation for something like this is `Light[4](Light(UBO[0].xyz, ...`. It's probably possible to work around this if there is a way to emit extra variables but TBH I haven't really seen a shader that copies an array out of the uniform buffers so the value is questionable...